File: tournaments/areIsomorphic/areIsomorphic.cs
bool areIsomorphic(int[][] array1, int[][] array2) {
    var rows1 = array1.Length;
    var rows2 = array2.Length;

    if (rows1 != rows2) {
        return false;
    }

    for (int i = 0; rows1 > i; ++i) {
        if (array1[i].Length != array2[i].Length) {
            return false;
        }
    }

    return true;
}

File: tournaments/areIsomorphic/areIsomorphic.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub areIsomorphic {
    my ($array1, $array2) = @_;

    unless (@$array1 == @$array2) {
        return 0;
    }

    for (my $i = 0; @$array1 > $i; ++$i) {
        unless (@{$$array1[$i]} == @{$$array2[$i]}) {
            return 0;
        }
    }

    return 1;
}
